api/v1alpha1/dnshealthcheckprobe_types.go

```diff
@@ -42,13 +41,13 @@ type DNSHealthCheckProbeSpec struct {
 	// +kubebuilder:validation:XValidation:rule="self in ['HTTP','HTTPS']",message="Only HTTP or HTTPS protocols are allowed"
 	Protocol Protocol `json:"protocol,omitempty"`
 	// Interval defines how frequently this probe should execute
-	Interval time.Duration `json:"interval,omitempty"`
+	Interval metav1.Duration `json:"interval,omitempty"`
 	// AdditionalHeadersRef refers to a secret that contains extra headers to send in the probe request, this is primarily useful if an authentication
 	// token is required by the endpoint.
 	AdditionalHeadersRef *AdditionalHeadersRef `json:"additionalHeadersRef,omitempty"`
 	// FailureThreshold is a limit of consecutive failures that must occur for a host to be considered unhealthy
 	// +kubebuilder:validation:XValidation:rule="self > 0",message="Failure threshold must be greater than 0"
-	FailureThreshold *int `json:"failureThreshold,omitempty"`
+	FailureThreshold int `json:"failureThreshold,omitempty"`
```

Comment:
same here the default will be 0

Reply:
This will only alow integers grater than 0